lib/relative.date.js

```diff
@@ -7,13 +7,28 @@ var noop = function noop () {};
 
 module.exports = function RelativeDate (date) {
   if (!(this instanceof RelativeDate)) return new RelativeDate(date);
-  date = date ? new Date(date) : new Date();
 
+  var lastTimeout = null;
   var listener = noop;
+
   this.onTick = function onTick (newListener) {
     listener = newListener;
   };
 
+  this.setDate = function setDate (newDate) {
+    date = newDate ? new Date(newDate) : new Date();
+    clearTick();
```

Comment:
I think clearing the previous timeout could be done closer to where we set the next (around line 40). So everytime `tick` runs it'll always clear the previous timeout regardless.